specs/fast-cli.md

```diff
@@ -0,0 +1,164 @@
+    - [Command Line](#command-line)
+        - [Initialization](#initialization)
+        - [Adding A Component](#adding-a-component)
+            - [Adding A Foundation Based Component](#adding-a-foundation-based-component)
+    - [Structure](#structure)
+        - [Create FAST Project Package](#create-fast-project-package)
+        - [FAST CLI Package](#fast-cli-package)
+    - [Dependencies](#dependencies)
+- [Implementation](#implementation)
+- [Documentation](#documentation)
+    - [System Requirements](#system-requirements)
+    - [Use Of The CLI](#use-of-the-cli)
+    - [How To Create Templates](#how-to-create-templates)
+- [Maintenance](#maintenance)
+    - [Testing](#testing)
+- [Roadmap](#roadmap)
+    - [MVP](#mvp)
+
+## Abstract
+
+This specification describes a CLI used for the creation and running of FAST based applications and component libraries called `@microsoft/fast-cli`, as well as a package leveraging the CLI and allowing `npm` to initialize it called `@microsoft/create-fast-project`. The CLI is intended to accelerate user workflows when using FAST, and the initialization package is intended to assist in a quick setup for projects such as applications and component libraries.
+
+## Use Cases
+
+**Creating a prototype**: Bob is a hobbyist illustrator and wants to create a FAST web component based application quickly to prototype a website for his work. He knows how to write HTML, CSS, and some JavaScript, but does not want to get bogged down with testing or type checking.
+
+**Creating a production application**: April is a developer who has been tasked with creating an application that must run efficiently and is maintainable. She wants a quick setup of an application for a production environment which should include type checking, testing and server side rendering.
+
+**Creating a component library**: Wilhelm is a UI developer, he wants to create a library of FAST web components for his company to use in their application. His companys current application has mismatched design concepts, so he has also been developing a design system and wants the components he creates to conform to it.
+
+## Requirements
+
+### Configuration
+
+The configuration of the project will occur during the setup using `npm init @microsoft/fast-project ./my-project` see [`npm init` documentation](https://docs.npmjs.com/cli/v6/commands/npm-init). This requires the user has npm >=6.0.0. Certain CLI commands can be used depending on the configuration, such as adding boilerplate components with `fast add --component`. The following configuration options will be determined during initialization.
+
+#### Project Type
+
+There will be two project types, `application` and `component-library`. When this is specified, this will allow the CLI to make assumptions about the project. The projects structure will change based on what type of project this is, additionally `webpack` will be added and configured if the project is an application.
```

Comment:
I think we need a third type here that is even more basic. IE, I want to create a single web component (per note about user stories)